fix(alphabetical): account for lost proxies

P0lip opened the pull request:
Fixes https://github.com/stoplightio/spectral/issues/979

Requires https://github.com/stoplightio/yaml/pull/40 and https://github.com/stoplightio/json/pull/43
**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

This trick should not be required once we have a different more suited json-ref-resolver library. :D

nulltoken commented on src/functions/__tests__/alphabetical.test.ts:
```diff
@@ -88,4 +101,34 @@ describe('alphabetical', () => {
+  test('is able to trap object again', () => {
+    const document = new Document(
+      `'404':
+'200':`,
```
@P0lip Can't this be put in a single line with a `\n`?

P0lip replied:
Yup!

P0lip commented:
In order to keep things going, going to merge it.
Just 2 little touches, one addressing comment re test from @nulltoken and the other adjusting the build process.
